app/assets/javascripts/discourse/app/components/passkey-login-button.js

```diff
@@ -0,0 +1,38 @@
+import Component from "@ember/component";
+import { action } from "@ember/object";
+import { ajax } from "discourse/lib/ajax";
+import { prepPasskeyCredential } from "discourse/lib/webauthn";
+import { inject as service } from "@ember/service";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+
+export default class PasskeyLoginButton extends Component {
+  @service dialog;
+  tagName = "";
+
+  @action
+  passkeyLogin() {
+    ajax("/session/passkey/challenge.json")
+      .then((response) => {
+        prepPasskeyCredential(response.challenge, (errorMessage) => {
+          this.dialog.alert(errorMessage);
+        }).then((credential) => {
+          ajax("/session/passkey/auth.json", {
+            type: "POST",
+            data: {
+              publicKeyCredential: credential,
+              timezone: moment.tz.guess(),
+            },
+          })
+            .then((result) => {
+              if (result && !result.error) {
+                window.location.reload();
```

Comment:
Just curious, why do we reload the window here?

Reply:
Good question. I added it for simplicity while working on a proof of concept. This likely needs to be changed. 

In the regular login form, we have some legacy code as well, I think. We submit a hidden form for most browser, except for Safary on iOS, where we do a redirect via `window.location.href = redirect`. 

See https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/app/components/modal/login.js#L239-L251